### PR TITLE
Commit to add mail sending of donation certificates

### PR DESCRIPTION
**Pull request by brennerthomas**
- Adding in the UI a modal to make settings for donation certificate more intuitive and cleaner
- Adding permissions that only global_admin can execute triggering of donation certificates and mail

### CODE REVIEW

File: admin/src/App.tsx [hunk 1]
```diff
@@ -142,6 +142,7 @@ export default function App() {
 					if (result?.values?.is_global_admin) {
 						setCollections(globalAdminCollections);
 						setCustomViews(publicCustomViews.concat(globalAdminCustomViews));
+						authController.setExtra({ isGlobalAdmin: true });
```
Comment (andrashee) [inline admin/src/App.tsx:145]: Neat implementation 👍

File: functions/src/donation_certificates/createDonationCertificatesFunction.ts [hunk 2]
```diff
@@ -4,14 +4,16 @@ import { useFirestore } from '../../../shared/src/firebase/firestoreAdmin';
...
-export interface CreateDonationCertificatesFunctionProps {
+export interface DonationCertificatesFunctionProps {
 	users: Entity<User>[];
 	year: number;
+	mailFlag: boolean;
 }
 
 export const createDonationCertificatesFunction = functions.https.onCall(
-	async ({ users, year }: CreateDonationCertificatesFunctionProps) => {
+	async ({ users, year, mailFlag }: DonationCertificatesFunctionProps) => {
```
Comment (andrashee) [inline functions/src/donation_certificates/createDonationCertificatesFunction.ts:16]: Shall we add an auth part here as well (e.g. in a follow-up PR)? Not showing the button in the admin is important for usability, but this function is still publicly callable afaik.

File: functions/src/donation_certificates/sendDonationCertificateMail.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
+const nodemailer = require("nodemailer");
+import { Entity, User } from '../../../shared/src/types';
+
+export const sendDonationCertificateMail = async (userEntity: Entity<User>, year: number, path: string) => {
+
+    let user = userEntity.values;
```
Comment (andrashee) [inline functions/src/donation_certificates/sendDonationCertificateMail.ts:6]: nit: shall we use const?

File: functions/src/donation_certificates/sendDonationCertificateMail.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    let translations = user.language === 'DE' ? GERMAN_TRANSLATIONS : (
+            user.language === 'FR' ? FRENCH_TRANSLATIONS : ENGLISH_TRANSLATIONS);
+
+    let transporter: any;
```
Comment (andrashee) [inline functions/src/donation_certificates/sendDonationCertificateMail.ts:11]: Shall we extract this to a reusable utility function? Could be in `/shared`?

**Review by andrashee — COMMENTED** [review]
Very nice!  
I will let @mkue comment on the react part.

One small thing: the email template doesn't seem to be valid html. The prettify jobs fail with:
```
Error:  shared/emails/transactional/donation-certificate-email-de.html: SyntaxError: Unexpected closing tag "span". It may happen when the tag has already been closed by another tag. For more info see https://www.w3.org/TR/html5/syntax.html#closing-elements-that-have-implied-end-tags (163:33)
[124](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:126)
Error:    161 |                                   <p style="margin: 0;">Ich sende dir anbei deine Spendenbescheinigung<!-- das Jahr {{year}}-->. Herzlichen Dank für dein Vertrauen und Engagement für weniger Armut und mehr Gerechtigkeit.
[125](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:127)
Error:    162 |
[126](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:128)
Error:  > 163 |                                 </span></p>
[127](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:129)
Error:        |                                 ^^^^^^^
[128](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:130)
Error:    164 |                                 </div>
[129](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:131)
Error:    165 |                               </td>
[130](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:132)
Error:    166 |                             </tr>
[131](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:133)

[132](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:134)
Error:  shared/emails/transactional/donation-certificate-email-en.html: SyntaxError: Unexpected closing tag "span". It may happen when the tag has already been closed by another tag. For more info see https://www.w3.org/TR/html5/syntax.html#closing-elements-that-have-implied-end-tags (163:33)
[133](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:135)
Error:    161 |                                   <p style="margin: 0;">I am sending you the donation receipt<!-- for the year {{year}}-->. Thank you for your trust and commitment to less poverty and more economic justice.
[134](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:136)
Error:    162 |
[135](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:137)
Error:  > 163 |                                 </span></p>
[136](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:138)
Error:        |                                 ^^^^^^^
[137](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:139)
Error:    164 |                                 </div>
[138](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:140)
Error:    165 |                               </td>
[139](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:141)
Error:    166 |                             </tr>
[140](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:142)

[141](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:143)
Error:  shared/emails/transactional/donation-certificate-email-fr.html: SyntaxError: Unexpected closing tag "span". It may happen when the tag has already been closed by another tag. For more info see https://www.w3.org/TR/html5/syntax.html#closing-elements-that-have-implied-end-tags (163:33)
[142](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:144)
Error:    161 |                                   <p style="margin: 0;">Je te envoie le reçu de don<!-- pour l'année {{year}}-->. Merci pour ta confiance et ton engagement pour moins de pauvreté et plus de justice.
[143](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:145)
Error:    162 |
[144](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:146)
Error:  > 163 |                                 </span></p>
[145](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:147)
Error:        |                                 ^^^^^^^
[146](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:148)
Error:    164 |                                 </div>
[147](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:149)
Error:    165 |                               </td>
[148](https://github.com/socialincome-san/public/actions/runs/3775481182/jobs/6418188389#step:3:150)
Error:    166 |                             </tr>
```

File: functions/src/donation_certificates/createDonationCertificatesFunction.ts (hunk 2 above)
Comment (mkue) [reply functions/src/donation_certificates/createDonationCertificatesFunction.ts]: Yes, we have to. I have the same issue in my other PR, #232.

Comment (mkue) [reply functions/src/donation_certificates/createDonationCertificatesFunction.ts]: How should I implement this? Make sure that the email from the token is an email of a global administrator?

**Review by mkue — APPROVED** [review]

**Comment by mkue** [comment]
Cool stuff @brennerthomas! The test email account setup is super convenient👍 

I updated your changes a bit and hopefully fixed the issue on production where we didn't have access to the assets folder.

**Pull request merged**